File: 143_ReorderList_BruteForce.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    void reorderList(ListNode* head) {
        
        ListNode *temp = head;
        vector<int> result;
        
        while(temp != nullptr)
        {
            result.push_back(temp->val);
            temp = temp->next;
        }
        
        int left = 0;
        int right = result.size() - 1;
        
        temp = head;
        
        while(left <= right)
        {
            temp->val = result[left];
            if(left != right)
            {
                temp->next->val = result[right];
                temp = temp->next->next;
            }

            left += 1;
            right -= 1;
        }
    }
};

File: 143_ReorderList_BruteForce.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        """
        Do not return anything, modify head in-place instead.
        """
        
        result = []
        
        dummy = head
        count = 0
    
        while(dummy):
            result.append(dummy.val)
            dummy = dummy.next
            count += 1
        
        #print(result)
        left = 0
        right = len(result) - 1
        
        dummy = head
        
        while left <= right:
            dummy.val = result[left]
            if left != right:
                dummy.next.val = result[right]
                dummy = dummy.next.next
            
            left += 1
            right -= 1
        
        
        
        
        
        